mmr/mmr.go

```diff
@@ -535,7 +552,7 @@ func takeWhileVec(v *[]types.Leaf, p func(types.Leaf) bool) []types.Leaf {
 	return vCopy[:]
 }
 
-func takeWhileVecUint64(v *[]uint64, p func(uint64) bool) []uint64 {
+func filterLeavesByPosition(v *[]uint64, p func(uint64) bool) []uint64 {
 	vCopy := *v
```

Comment:
why do you need to do a full copy? isn't there a way to keep track of the indices that you're interested in, and at the end select the subslice you want?

Reply:
It's possible to use the dereferenced value without first assigning it to a variable. I only did that to make it cleaner